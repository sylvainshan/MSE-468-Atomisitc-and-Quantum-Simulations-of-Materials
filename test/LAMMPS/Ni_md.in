# LAMMPS input script for a molecular dynamics simulation of Ni using an EAM potential

# Variables
variable supercell equal 5
variable timestep equal 0.001
variable stride equal 1000

# Initialize simulation
clear
units metal
dimension 3
boundary p p p
atom_style atomic
atom_modify map hash

# Create atom definition
lattice custom 3.5400 &
	a1 1 0 0 & 
	a2 0 1 0 &
	a3 0 0 1 &
	basis 0.0000 0.0000 0.0000 &
	basis 0.0000 0.5000 0.5000 &
	basis 0.5000 0.0000 0.5000 &
	basis 0.5000 0.5000 0.0000 

region myreg block 0 ${supercell} 0 ${supercell} 0 ${supercell}
create_box 1 myreg
create_atoms 1 box
mass 1 58.6934

# Set interaction potential
pair_style eam
pair_coeff * * Ni_u3.eam

# Define simulation parameters
timestep ${timestep}
# Following would be return in the output file
thermo ${stride}
thermo_style custom time temp pe ke press

# Dumping positions (scaled) and velocities

dump dp1 all atom ${stride} positions.lammpstrj
dump dp2 all custom ${stride} velocities.lammpstrj id type vx vy vz
dump_modify dp2 format line "%d %d %g %g %g" 

# Set initial temperature and velocity
velocity all create 1000.0 87287 loop geom

# Run equilibration
# last quantity is T damp which should be about 100 times timestep but using variables
# like $(100.0*${timestep}) or $(100.0*timestep) doesn't work, so for now it is hardcoded
fix 1 all nvt temp 1000.0 1000.0 0.1
run 5000

# Run production simulation
unfix 1
fix 2 all nvt temp 1000.0 1000.0 0.1 
run 30000

# Dumping recovery file
write_data recovery.data
