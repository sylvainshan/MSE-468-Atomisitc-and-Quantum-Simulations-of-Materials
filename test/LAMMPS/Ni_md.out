LAMMPS (7 Feb 2024 - Development - patch_7Feb2024_update1-35-g81609d04e3)
Lattice spacing in x,y,z = 3.54 3.54 3.54
Created orthogonal box = (0 0 0) to (17.7 17.7 17.7)
  1 by 1 by 2 MPI processor grid
Created 500 atoms
  using lattice units in orthogonal box = (0 0 0) to (17.7 17.7 17.7)
  create_atoms CPU = 0.001 seconds
Reading eam potential file Ni_u3.eam with DATE: 2007-06-11
Neighbor list info ...
  update: every = 1 steps, delay = 0 steps, check = yes
  max neighbors/atom: 2000, page size: 100000
  master list distance cutoff = 6.8
  ghost atom cutoff = 6.8
  binsize = 3.4, bins = 6 6 6
  1 neighbor lists, perpetual/occasional/extra = 1 0 0
  (1) pair eam, perpetual
      attributes: half, newton on
      pair build: half/bin/atomonly/newton
      stencil: half/bin/3d
      bin: standard
Setting up Verlet run ...
  Unit style    : metal
  Current step  : 0
  Time step     : 0.001
Per MPI rank memory allocation (min/avg/max) = 3.898 | 3.898 | 3.898 Mbytes
     Time           Temp          PotEng         KinEng         Press     
 0              1000          -2224.1234      64.500812     -17188.194    
 1              941.11804     -2162.5755      60.702878      44196.603    
 2              1001.9293     -2156.9152      64.625251      50326.393    
 3              932.01717     -2157.6147      60.115865      48273.673    
 4              1010.0137     -2157.6541      65.146707      49607.907    
 5              988.37354     -2163.3269      63.750897      44033.502    
Loop time of 1.4083 on 2 procs for 5000 steps with 500 atoms

Performance: 306.753 ns/day, 0.078 hours/ns, 3550.380 timesteps/s, 1.775 Matom-step/s
40.1% CPU use with 2 MPI tasks x no OpenMP threads

MPI task timing breakdown:
Section |  min time  |  avg time  |  max time  |%varavg| %total
---------------------------------------------------------------
Pair    | 1.1644     | 1.191      | 1.2176     |   2.4 | 84.57
Neigh   | 0          | 0          | 0          |   0.0 |  0.00
Comm    | 0.10362    | 0.1387     | 0.17378    |   9.4 |  9.85
Output  | 0.03256    | 0.034251   | 0.035942   |   0.9 |  2.43
Modify  | 0.020403   | 0.029483   | 0.038563   |   5.3 |  2.09
Other   |            | 0.01483    |            |       |  1.05

Nlocal:            250 ave         250 max         250 min
Histogram: 2 0 0 0 0 0 0 0 0 0
Nghost:           1484 ave        1484 max        1484 min
Histogram: 2 0 0 0 0 0 0 0 0 0
Neighs:          16750 ave       17300 max       16200 min
Histogram: 1 0 0 0 0 0 0 0 0 1

Total # of neighbors = 33500
Ave neighs/atom = 67
Neighbor list builds = 0
Dangerous builds = 0
Setting up Verlet run ...
  Unit style    : metal
  Current step  : 5000
  Time step     : 0.001
Per MPI rank memory allocation (min/avg/max) = 6.167 | 6.167 | 6.167 Mbytes
     Time           Temp          PotEng         KinEng         Press     
 5              988.37354     -2163.3269      63.750897      44033.502    
 6              1027.9672     -2161.0345      66.304716      46447.506    
 7              989.01966     -2163.7463      63.792572      43074.868    
 8              979.11481     -2158.4905      63.153701      48650.659    
 9              1034.7028     -2159.6301      66.739172      47503.198    
 10             970.03464     -2161.9549      62.568022      44710.636    
 11             991.14248     -2159.1911      63.929495      47940.984    
 12             1020.3113     -2159.3569      65.81091       47812.263    
 13             1001.813      -2158.9418      64.617754      48761.889    
 14             975.59234     -2161.4221      62.926499      45408.317    
 15             1005.0603     -2157.7382      64.827205      49527.138    
 16             958.0922      -2162.373       61.797725      44002.041    
 17             1006.6016     -2162.0425      64.926624      45559.745    
 18             1059.5972     -2162.7362      68.344879      45586.152    
 19             976.21014     -2166.2714      62.966347      40609.651    
 20             1034.2957     -2160.9728      66.712911      47202.078    
 21             1030.601      -2159.3861      66.474605      48258.988    
 22             1005.2061     -2157.5039      64.836611      50173.384    
 23             958.27464     -2163.8015      61.809493      42910.917    
 24             1005.9139     -2158.1204      64.882265      49506.215    
 25             1008.2058     -2160.1294      65.030091      47310.044    
 26             988.16233     -2161.4694      63.737273      45787.697    
 27             1054.2807     -2159.7745      68.001962      48314.42     
 28             1009.9921     -2161.5768      65.145311      45686.016    
 29             1021.2427     -2158.9664      65.870982      48848.417    
 30             993.29172     -2160.7228      64.068123      45930.462    
 31             990.36769     -2162.2874      63.87952       45322.361    
 32             997.22021     -2158.2097      64.321513      48729.905    
 33             982.14311     -2162.3501      63.349028      44777.744    
 34             991.80756     -2157.8192      63.972393      49441.732    
 35             951.36533     -2160.7426      61.363836      45933.274    
Loop time of 11.8573 on 2 procs for 30000 steps with 500 atoms

Performance: 218.600 ns/day, 0.110 hours/ns, 2530.097 timesteps/s, 1.265 Matom-step/s
49.8% CPU use with 2 MPI tasks x no OpenMP threads

MPI task timing breakdown:
Section |  min time  |  avg time  |  max time  |%varavg| %total
---------------------------------------------------------------
Pair    | 10.15      | 10.367     | 10.583     |   6.7 | 87.43
Neigh   | 0          | 0          | 0          |   0.0 |  0.00
Comm    | 0.88908    | 1.105      | 1.321      |  20.5 |  9.32
Output  | 0.10037    | 0.10731    | 0.11424    |   2.1 |  0.90
Modify  | 0.1641     | 0.1689     | 0.17369    |   1.2 |  1.42
Other   |            | 0.1092     |            |       |  0.92

Nlocal:            250 ave         251 max         249 min
Histogram: 1 0 0 0 0 0 0 0 0 1
Nghost:           1484 ave        1485 max        1483 min
Histogram: 1 0 0 0 0 0 0 0 0 1
Neighs:          15939 ave       16473 max       15405 min
Histogram: 1 0 0 0 0 0 0 0 0 1

Total # of neighbors = 31878
Ave neighs/atom = 63.756
Neighbor list builds = 0
Dangerous builds = 0
System init for write_data ...
Total wall time: 0:00:13
